rc/filetype/css.kak

```diff
@@ -32,28 +33,78 @@ provide-module css %[
+
+  # Units
+  # ‾‾‾‾‾
+  # generated from the URL & <code> below
+  # includes #rgb, #rrggbb, #rrggbbaa as color values {3,8}
+  # 
+  # https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units
+  units='% cap ch cm deg dpcm dpi dppx em ex grad Hz ic in kHz lh mm ms pc pt px Q rad rem rlh s turn vb vh vi vmax vmin vw x'
+
+  logical_ops='and not only from to'
+  keywords='!important auto inherit initial unset none'
+  media_types='all print screen speech'
+
+  # easing_re='linear|ease(-(in-out|in|out))?|step-(start|end)'
+
+  # combinators='+ > ~ ||'
+  # attribute_op='= ~= |= ^= $= *='
+
+  join() { eval set -- $1; IFS="|"; echo "$*"; }
+
+  # Selectors
+  # ‾‾‾‾‾‾‾‾‾
+  # universal: *, ns|*, *|*, |*
+  # class/id: .class, #id
+  # type: element
+  # attr: [attr=val]
+
+  # order below matters
+  printf %s "
+  add-highlighter shared/css/code/tag_selectors regex \b($(join "${html_tags}"))((:[a-z:])|[\h.#]) 1:keyword
 
-add-highlighter shared/css/declaration/base default-region group
-add-highlighter shared/css/declaration/double_string region '"' (?<!\\)(\\\\)*" fill string
-add-highlighter shared/css/declaration/single_string region "'" "'"             fill string
-add-highlighter shared/css/declaration/comment region /[*] [*]/ fill comment
+  add-highlighter shared/css/code/functional_notation regex ([a-zA-Z0-9-_]+[a-zA-Z0-9])\( 1:keyword
 
-# https://developer.mozilla.org/en-US/docs/Web/CSS/length
-add-highlighter shared/css/declaration/base/ regex (#[0-9A-Fa-f]+)|((\d*\.)?\d+(ch|cm|em|ex|mm|pc|pt|px|rem|vh|vmax|vmin|vw)) 0:value
+  add-highlighter shared/css/code/logical_operators regex (\b($(join "${logical_ops}"))\b|$(join "${keywords}")) 1:keyword 1:+i
 
-add-highlighter shared/css/declaration/base/ regex ([A-Za-z][A-Za-z0-9_-]*)\h*: 1:keyword
-add-highlighter shared/css/declaration/base/ regex :(before|after) 0:attribute
-add-highlighter shared/css/declaration/base/ regex !important 0:keyword
+  add-highlighter shared/css/code/media_types regex \b($(join "${media_types}"))\b 1:+i
 
-# element#id element.class
-# universal selector
-add-highlighter shared/css/selector/ regex         [A-Za-z][A-Za-z0-9_-]* 0:keyword
-add-highlighter shared/css/selector/ regex [*]|[#.][A-Za-z][A-Za-z0-9_-]* 0:variable
+  # (after functional notation as they may contain paranthesis)
+  add-highlighter shared/css/code/psudo regex (:{1,2})([a-z-]+) 2:attribute 2:+a
```

Comment:
Typo: "pseudo"

You might as well include the `:` or `::` in the highlighting, since the punctuation is highlighted in `.class` and `#id` selectors.

Why `+a`? That means other styles won't be able to override attributes (bold, underline, etc.) but you don't know what the attributes of the `attribute` face are, so this seems brittle.

Reply:
Other editors do not include the `:` and `::` highlighted, again tried to follow existing styles, rather than just create my own.

Regarding `+a`... there must of been a reason I added this initially, have removed it and don't see it making a difference so, I'll commit that as well.

Reply:
I figure `:` and `::` are as much a part of the "spelling" of a (pseudo)class as `.` or `#` are part of the "spelling" of class or ID selectors. It may be that there's a good syntactical reason to treat them separately, but I think Kakoune being consistent with itself is more important than being consistent with some other editor.

Reply:
Ok, keeping them as part of spelling -- committed and pushed.